### PR TITLE
Add pre-commit check-spelling hook

### PR DESCRIPTION
**Pull request by l0uden**
## Description

Pre-commit `check-spelling` hook for words `colour` and `visualisation`

## Checklist

- [x] I have not referenced individuals, products or companies in any commits, directly or indirectly
- [x] I have not added data or restricted code in any commits, directly or indirectly
- [ ] I have updated the docstring of any public function/class/model changed
- [ ] I have added the PR number to the change description in the changelog fragment, e.g. `Enable feature XXX ([#1](https://github.com/mckinsey/vizro/pull/1))` (if applicable)

## Types of changes

- [x] Docs/refactoring (non-breaking change which improves codebase)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Review by huong-li-nguyen — APPROVED** [review]
In general good-to-go for me, @l0uden could you just double-check that is is not possible with the already existing pre-commit hook codespell? If this is not supported there and we need our custom script, fine to merge as is. Otherwise I would prefer to just configure the existing pre-commit hook

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -17,6 +17,11 @@ repos:
         entry: python tools/check_for_datafiles.py
         language: python
         pass_filenames: false
+      - id: check-spelling
+        name: check-spelling
+        entry: tools/find_forbidden_words_in_repo.sh
+        language: script
+        pass_filenames: false
```
[the remark this answers is not in the document]
Comment (l0uden) [reply .pre-commit-config.yaml:24]: I will check

**Comment by l0uden** [comment]
> In general good-to-go for me, @l0uden could you just double-check that is is not possible with the already existing pre-commit hook codespell? If this is not supported there and we need our custom script, fine to merge as is. Otherwise I would prefer to just configure the existing pre-commit hook

I thought we cannot use not python hooks from [here](https://pre-commit.com/hooks.html), but it looks like we can. Because if hook type is not python, it has lots of spelling variations

File: vizro-core/changelog.d/20230913_134437_alexey_snigir_check_spelling_hook.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+<!--
+### Removed
+
+- A bullet item for the Removed category.
+
+-->
+
+### Added
+
+- Pre-commit `check-spelling` hook for words `colour` and `visualisation`.
```
Comment (maxschulz-COL) [inline vizro-core/changelog.d/20230913_134437_alexey_snigir_check_spelling_hook.md:16]: I think this is also not a change we need to notify our users about.

Comment (l0uden) [reply vizro-core/changelog.d/20230913_134437_alexey_snigir_check_spelling_hook.md]: Deleted

File: .pre-commit-config.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (l0uden) [reply .pre-commit-config.yaml:24]: For what I've investigated codespell doesn't support forbidden words check

**Review by huong-li-nguyen — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (huong-li-nguyen) [reply .pre-commit-config.yaml:24]: Thanks for checking though! :)

**Review by maxschulz-COL — APPROVED** [review]
Approving as it looks good to me. Remind me though, what was the conversation regarding using a hook? I thought we wanted to go down that path?

**Comment by l0uden** [comment]
> Approving as it looks good to me. Remind me though, what was the conversation regarding using a hook? I thought we wanted to go down that path?

Unfortunately I didn't found the hook with such a simple functionality

**Pull request merged**